presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -271,34 +276,15 @@ public static LongWriteFunction dateWriteFunction()
      * {@link #timeColumnMapping} instead.
      */
     @Deprecated
-    public static ColumnMapping timeColumnMappingUsingSqlTime(ConnectorSession session)
+    public static ColumnMapping timeColumnMappingUsingSqlTime()
     {
-        if (session.isLegacyTimestamp()) {
-            ZoneId sessionZone = ZoneId.of(session.getTimeZoneKey().getId());
-            return ColumnMapping.longMapping(
-                    TIME,
-                    (resultSet, columnIndex) -> {
-                        Time time = resultSet.getTime(columnIndex);
-                        return toPrestoLegacyTimestamp(toLocalTime(time).atDate(LocalDate.ofEpochDay(0)), sessionZone);
-                    },
-                    timeWriteFunctionUsingSqlTime(session));
-        }
-
         return ColumnMapping.longMapping(
                 TIME,
                 (resultSet, columnIndex) -> {
                     Time time = resultSet.getTime(columnIndex);
-                    return NANOSECONDS.toMillis(toLocalTime(time).toNanoOfDay());
+                    return time.getTime() * PICOSECONDS_PER_MILLISECOND % PICOSECONDS_PER_DAY;
```

Comment:
`time.getTime()` is a timestamp that, if interpreted in JVM zone, refers to hour/minute/second represented.
to obtain number of millis/nanos since midnight you need to  `time.toLocalTime().toNanoOfDay()`

Reply:
`getTime` is in milliseconds since the epoch, (and `java.sql.Time` specifies that the hidden day/month/year should be set to the epoch).

`toLocalTime` only copies hours/minutes/seconds.

The JDBC driver's handling for time zones for types without zones can affect the correctness of these conversions, though.

Reply:
@jirassimok right, `time.toLocalTime()` loses subsecond information. 
My point is -- `time.getTime()` is not the value we want. It's different from the value we want by JVM zone's offset on 1970-01-01